drivers/hmis/spec/requests/hmis/client_alert_spec.rb

```diff
@@ -0,0 +1,56 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+require 'rails_helper'
+require_relative 'login_and_permissions'
+require_relative '../../support/hmis_base_setup'
+
+RSpec.describe Hmis::GraphqlController, type: :request do
+  before(:all) do
+    cleanup_test_environment
+  end
+  after(:all) do
+    cleanup_test_environment
+  end
+
+  before(:each) do
+    hmis_login(user)
+  end
+
+  include_context 'hmis base setup'
+
+  let!(:access_control) { create_access_control(hmis_user, p1) }
+
+  describe 'querying a client with alerts' do
+    let(:query) do
+      <<~GRAPHQL
+        query Client($id: ID!) {
+          client(id: $id) {
+            id
+            alerts {
+              id
+              note
+            }
+          }
+        }
+      GRAPHQL
+    end
+
+    let!(:c1) { create :hmis_hud_client, data_source: ds1 }
+    let!(:a1) { create :hmis_client_alert, created_by: hmis_user, client: c1 }
+
+    it 'should return the client with an alert' do
```

Comment:
lets add a test case for a user that does not have permission to view alerts. it should return no alerts even if they exist

Reply:
Good plan, let me know what you think about the test case that I added. I don't know if it would ever really be the case that a user has permission to view a client but not that client's alerts -- maybe more likely they would either be able to view a client and their alerts, or not view the client at all? -- I just wrote it this way so the test is testing the actual permission `:can_view_client_alerts` instead of something less granular. But let me know if I should take a different approach with this!

Reply:
Test looks good to me!

I do think it's likely that a user may have access to view a Client but not their Alerts. I can think of at least two scenarios:
1. The installation doesn't want to use Client Alerts at all, so we don't give the alert-viewing permission to anyone
2. A setup where Providers can see all Clients, but they can only see Client Alerts for the clients that are enrolled at their programs